packages/wonder-blocks-timing/src/hooks/__tests__/use-timeout.test.ts

```diff
@@ -7,129 +8,484 @@ describe("useTimeout", () => {
-        );
+        const {result} = renderHook(() => useTimeout(() => {}, 1000));
 
         // Act
-        rerender({active: true});
-        jest.advanceTimersByTime(501);
 
         // Assert
-        expect(action).toHaveBeenCalled();
+        expect(result.current.isSet).toBe(true);
     });
 
-    it("should not fire after the timeout if 'active' changes to false", () => {
-        // Arrange
+    it("should call the action before unmounting", () => {
         const action = jest.fn();
-        const {rerender} = renderHook(
-            ({active}: any) => useTimeout(action, 500, active),
-            {initialProps: {active: true}},
+        const {unmount} = renderHook(() =>
+            useTimeout(action, 1000, {
+                clearPolicy: ClearPolicy.Resolve,
+            }),
         );
 
-        // Act
-        rerender({active: false});
-        jest.advanceTimersByTime(501);
+        act(() => {
+            unmount();
+        });
 
-        // Assert
-        expect(action).not.toHaveBeenCalled();
+        expect(action).toHaveBeenCalled();
     });
 
-    it("should reset the timeout if 'timeoutMs' is changes", () => {
+    it("should call the current action", () => {
         // Arrange
-        const action = jest.fn();
-
-        // Act
+        const action1 = jest.fn();
+        const action2 = jest.fn();
         const {rerender} = renderHook(
-            ({timeoutMs}: any) => useTimeout(action, timeoutMs, true),
-            {initialProps: {timeoutMs: 500}},
+            ({action}: any) => useTimeout(action, 500),
+            {
+                initialProps: {action: action1},
+            },
         );
-        rerender({timeoutMs: 1000});
+
+        // Act
+        rerender({action: action2});
         jest.advanceTimersByTime(501);
 
         // Assert
-        expect(action).not.toHaveBeenCalled();
-        jest.advanceTimersByTime(1001);
-        expect(action).toHaveBeenCalled();
+        expect(action2).toHaveBeenCalledTimes(1);
     });
 
-    it("should not reset the timeout if 'action' changes", () => {
+    it("should only call setTimeout once even if action changes", () => {
         // Arrange
+        const timeoutSpy = jest.spyOn(global, "setTimeout");
         const action1 = jest.fn();
         const action2 = jest.fn();
-        const timeoutSpy = jest.spyOn(window, "setTimeout");
-
-        // Act
         const {rerender} = renderHook(
-            ({action}: any) => useTimeout(action, 500, true),
-            {initialProps: {action: action1}},
+            ({action}: any) => useTimeout(action, 500),
+            {
+                initialProps: {action: action1},
+            },
         );
         // NOTE: For some reason setTimeout is called twice by the time we get
         // here.  I've verified that it only gets called once inside the hook
         // so something else must be calling it.
         const callCount = timeoutSpy.mock.calls.length;
```

Comment:
**question:** Is this still the case after your changes?

Reply:
Not sure. I can check.